validate.js

```diff
@@ -44,40 +52,34 @@ export default class Validate extends Validates {
    * @private
    */
   get validates() {
-    // Prefer props over state.
-    const { validates = this.state.validates } = this.props;
-    return validates;
+    return this.props.validates || this.state.validates;
   }
 
   /**
-   * Get the child context.
+   * Child validity change handler.
    *
-   * @returns {Object} The child context.
+   * @param {String} name Identifier for the field whose validity changed.
+   * @param {Validity} isValid The field's current validity.
+   * @private
    */
-  getChildContext() {
-    return {
-      /**
-       * Child validity change handler.
-       *
-       * @param {String} name Identifier for the field whose validity changed.
-       * @param {Validity} isValid The field's current validity.
-       */
-      onValidChange: (name, isValid) => {
-        const { valids } = this.state;
-        const { validate } = this.props;
-
-        if (isValid === undef) {
-          delete valids[name];
-        } else {
-          valids[name] = isValid;
-        }
-        this.setState({ valids, validates: validate(valids) });
-      }
-    };
+  processValidChange(name, isValid) {
+    const { valids } = this.state;
```

Comment:
does that need to be shallow cloned? you're technically modifying the object that's already in state and putting the same object reference back in, which might be okay. i guess you only really need to process the state change if `validates` changes?

Reply:
Maybe worth a comment so it doesn't look like a bug?

Reply:
Good find, this was still the original code, so it works, but i'll make clone of the object.

Reply:
Good find, this was still the original code, so it works, but i'll make clone of the object.

Reply:
Already wasted too much time on this, we can fix this in another sprint :joy: